packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1269,11 +1271,12 @@ class _TabBarViewState extends State<TabBarView> {
 
   void _updateChildren() {
     _children = widget.children;
-    _childrenWithKey = KeyedSubtree.ensureUniqueKeysForList(widget.children);
+    _originalChildrenWithKey = KeyedSubtree.ensureUniqueKeysForList(widget.children);
+    _childrenWithKey = _originalChildrenWithKey;
   }
 
   void _handleTabControllerAnimationTick() {
-    if (_warpUnderwayCount > 0 || !_controller.indexIsChanging)
+    if (_warpUnderwayCount > 0 && _controller.index == _currentIndex)
```

Comment:
is this still needed?

Reply:
Yes, because while ```_warpToCurrentIndex()``` is running and user clicks another tab, we enter ```_handleTabControllerAnimationTick()``` and we need to allow it to go all the way to ```_warpToCurrentIndex()``` again. This is the case when ```_controller.index``` and ```_currentIndex``` are different.

This wasn't needed before because all the animations had same duration, so even without the condition I added, prior animation would finish before the new animation which is triggered by tab click, so it eventually reached ```_warpToCurrentIndex()``` again.